File: Data Structures & Algorithms/combination-target-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */ 
    combinationSum(nums, target) { 
        const result = [] 
        this.rec(nums, [], target, result, 0) 
        return result
    } 

    rec(nums, cur, target, res, level) {
        let total = cur.reduce((acc, val) => acc + val, 0) 

        if (total > target) {
            return
        } else if (total === target) {
            res.push(cur)
        }  

        for (let i = level; i < nums.length; i ++) {
            let num = nums[i] 
            this.rec(nums, [...cur, num], target, res, i)
        }
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-3.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        curSum = 0 
        res = len(nums) + 1 

        l = 0  
        for r in range(len(nums)): 
            cur = nums[r] 
            curSum += cur 

            while curSum >= target:
                curLen = r - l + 1 
                res = min(curLen, res) 

                deleted = nums[l] 
                curSum -= deleted 
                l += 1

        return res if res != len(nums) + 1 else 0
